service/history/mutableStateBuilder.go

```diff
@@ -1344,8 +1346,8 @@ func (e *mutableStateBuilder) ReplicateWorkflowExecutionStartedEvent(
 	return nil
 }
 
-func (e *mutableStateBuilder) AddDecisionTaskScheduledEvent() (*decisionInfo, error) {
-
+// originalScheduledTimestamp is to record the first scheduled decision during decision heartbeat.
+func (e *mutableStateBuilder) AddDecisionTaskScheduledEventAsHeartbeat(originalScheduledTimestamp int64) (*decisionInfo, error) {
```

Comment:
```
AddDecisionTaskScheduledEventAsHeartbeat(
    originalScheduledTimestamp int64,
) (*decisionInfo, error)
```

Reply:
Why? does it look very long here?

Reply:
1. we had similar conversation during the standup
2. yes, it is too long
3. i have already converted most of functions in mutable state builder